src/person.cpp

```diff
@@ -408,6 +425,11 @@ static uint16 EncodeWalk(const WalkInformation *wi)
 			return (2 << 12) | i;
 		}
 	}
+	for (int i = 0; i < 4; i++) {
+		if (wi == _handyman_water[i]) {
+			return (2 << 12) | (1 << 8) | i;
```

Comment:
It's probably useful at some point to fold encoding and decoding these values into some functions, so you have a name what you're doing here, and a name what information you extract.

This also helps if you ever want to change the layout of these values, as finding all literal operations using magic numbers is a nightmare to say the least.

While thinking about it, you could fold the value in a struct, like
``` cpp
struct Thing {
   private int value;
   public setValue(int field1, int field2, ..) { .. }  // Constructor would work too of course.
   public getField1() { return ...; }
   ...
};
```
This should be as large as the number itself, since there are no `virtual` methods. It's also type-safe, you cannot inject a different value in the decoding function.